src/coreclr/jit/fgbasic.cpp

```diff
@@ -368,6 +368,16 @@ void Compiler::fgRemoveEhfSuccessor(BasicBlock* block, const unsigned succIndex)
                   (succCount - succIndex - 1) * sizeof(FlowEdge*));
     }
 
+    // If the block has other successors, distribute the removed edge's likelihood among the remaining successor edges.
+    if (succCount > 1)
+    {
+        const weight_t likelihoodIncrease = succEdge->getLikelihood() / (succCount - 1);
```

Comment:
Shouldn't these proportionally scale up?

Say there are 3 successors with likelihoods A 0.1, B 0.1, C 0.8. We remove A. Then we should have B = 0.1111..., C = 0.8888... 

With these changes we'd get B = 0.15, C = 0.85, so the relative likelihood of B would increase.

Generally $p_{i,new} = p_{i,old} / (1 - p_{removed})$, unless $p_{removed}$ is 1.0 or close to 1.0, in which case equal distribution seems ok.

Reply:
Good point; fixed